x-pack/plugins/security_solution/public/common/components/data_table/index.tsx

```diff
@@ -283,45 +311,47 @@ export const DataTableComponent = React.memo<DataTableProps>(
 
     const columnsWithCellActions: EuiDataGridColumn[] = useMemo(
       () =>
-        columnHeaders.map((header) => {
-          const buildAction = (dataTableCellAction: DataTableCellAction) =>
-            dataTableCellAction({
-              browserFields,
-              data: data.map((row) => row.data),
-              ecsData: data.map((row) => row.ecs),
-              header: columnHeaders.find((h) => h.id === header.id),
-              pageSize: pagination.pageSize,
-              scopeId: id,
-              closeCellPopover: dataGridRef.current?.closeCellPopover,
-            });
-          return {
-            ...header,
-            actions: {
-              ...header.actions,
-              additional: [
-                {
-                  iconType: 'cross',
-                  label: REMOVE_COLUMN,
-                  onClick: () => {
-                    dispatch(dataTableActions.removeColumn({ id, columnId: header.id }));
-                  },
-                  size: 'xs',
+        !isEventRenderedView
```

Comment:
This _useMemo_ implementation is getting pretty hard to read. If we wanted to be strict with [our styleguide](https://github.com/elastic/kibana/blob/main/STYLEGUIDE.mdx#only-use-ternary-operators-for-small-simple-code) we could just have an exit case at the top returning `columnHeaders` if `isEventRenderedView` is _true_, and then do the map. Just a NIT